lib/libmbedtls/mbedtls/library/bignum.c

```diff
@@ -1802,13 +1804,21 @@ int mbedtls_mpi_exp_mod( mbedtls_mpi *X, const mbedtls_mpi *A,
     if( mbedtls_mpi_cmp_int( E, 0 ) < 0 )
         return( MBEDTLS_ERR_MPI_BAD_INPUT_DATA );
 
+    W = mempool_alloc( mbedtls_mpi_mempool,
+                       sizeof( mbedtls_mpi ) * array_size_W );
+    if( W == NULL )
+    {
+        ret = MBEDTLS_ERR_MPI_ALLOC_FAILED;
```

Comment:
Prefer `return MBEDTLS_ERR_MPI_ALLOC_FAILED;` straight otherwise `mbedtls_mpi_free()` is called over uninited `T` and `Apos`.

Reply:
Good catch, I'll fix.